lib/class-wp-theme-json-gutenberg.php

```diff
@@ -551,6 +560,12 @@ private static function compute_style_properties( $styles ) {
 		foreach ( self::PROPERTIES_METADATA as $css_property => $value_path ) {
 			$value = self::get_property_value( $styles, $value_path );
 
+			if ( in_array( $css_property, self::PROTECTED_PROPERTIES ) ) {
+				if ( ! _wp_array_get( $settings, $value_path, false ) ) {
```

Comment:
I think we'll want to `null` check this in order to be consistent with #34491, which makes the block gap styles explicitly _opt-out_ rather than opt-in; according to that PR setting the `blockGap` setting to `false` should turn off the block support UI but _not_ the global gap styles.

Reply:
Of course, the drawback to explicitly checking for `null` is that we'll end up with styling regressions on blocks themes that don't opt in to the block support, but _do_ opt in to the global block gap styling (ie `blockGap: false` or omitting the setting entirely).

I think that just might be a limitation of the unified variable approach 🤔